chrome/content/zotero/xpcom/itemPaneManager.js

```diff
@@ -0,0 +1,340 @@
+ * @property {boolean} editable - Whether the section is in edit mode
+ * @property {(l10nArgs: string) => void} setL10nArgs - Set l10n args for section header
+ * @property {(l10nArgs: string) => void} setEnabled - Set pane enabled state
+ * @property {(summary: string) => void} setSectionSummary - Set pane section summary
+ * @property {(buttonType: string, options: {disabled?: boolean; hidden?: boolean}) => void} setSectionButtonStatus - Set pane section button status
+ * @typedef SectionHookArgs
+ * @type {SectionBasicHookArgs & SectionUIHookArgs}
+ * @typedef  {SectionHookArgs & { refresh: () => Promise<void> }} SectionInitHookArgs
+ * A `refresh` is exposed to plugins to allows plugins to refresh the section when necessary,
+ * e.g. item modify notifier callback. Note that calling `refresh` during initialization
+ * have no effect.
+ * @typedef  {SectionHookArgs & { newData: Zotero.Item }} SectionDataChangeHookArgs
+ * @typedef  {SectionHookArgs & { event: Event }} SectionEventHookArgs
+ * @typedef ItemDetailsSectionOptions
+ * @type {object}
+ * @property {string} paneID - Unique pane ID
+ * @property {string} pluginID - Set plugin ID to auto remove section when plugin is disabled/removed
+ * @property {SectionL10n & SectionIcon} head - Header options. Icon should be 16*16 and `label` need to be localized
+ * @property {SectionL10n & SectionIcon} sidenav - Sidenav options. Icon should be 20*20 and `tooltiptext` need to be localized
+ * @property {string} [fragment] - Pane fragment as string
+ * @property {(props: SectionInitHookArgs) => void} [onInit]
+ * Lifecycle hook called when section is initialized.
+ * You can use destructuring assignment to get the props:
+ * ```js
+ * onInit({ paneID, doc, body, item, editable, tabType, setL10nArgs, setEnabled,
+ *          setSectionSummary, setSectionButtonStatus, refresh }) {
+ *   // Your code here
+ * }
+ * ```
+ *
+ * Do:
+ * 1. Initialize data if necessary
+ * 2. Set up hooks, e.g. notifier callback
+ *
+ * Don't:
+ * 1. Render/refresh UI
+ * @property {(props: SectionBasicHookArgs) => void} [onDestroy]
+ * Lifecycle hook called when section is destroyed
+ *
+ * Do:
+ * 1. Remove data and release resource
+ * 2. Remove hooks, e.g. notifier callback
+ *
+ * Don't:
+ * 1. Render/refresh UI
+ * @property {(props: SectionDataChangeHookArgs) => boolean} [onItemChange]
+ * Lifecycle hook called when section's item change received
+ *
+ * Return `false` to abort the data change, otherwise the data change will be applied
+ * (stored internally and can be accessed by `props.item` in other hooks).
+ *
+ * Do:
+ * 1. Validate and decide whether to accept the ne data change
+ * 2. Update the section enabled state with `props.setEnabled`. For example, if the section
+ *   is only enabled in the readers, you can use:
+ * ```js
+ * onDataChange({ newData, setEnabled }) {
+ *   if (newData.type === "tabType") {
+ *     setEnabled(newData.value === "reader");
+ *   }
+ *   return true;
```

Comment:
The example document says "return false to cancel the change". What does that mean? Cancel what change?

Reply:
Oh, and above too:

> Return `false` to abort the data change, otherwise the data change will be applied
> (stored internally and can be accessed by `props.item` in other hooks).

I don't understand what this means. Isn't this just notifying the section that we've switched to a new item? How can that be aborted?

Reply:
Oh, yes. From the very beginning, I implemented this part following the way we do in our sections. For example, in the attachment box, the `item` is only set when it's an attachment, otherwise the `this._item` is not updated. So same here, it can be aborted, which means if you return false the `this._item` will not be updated.

Looking back, I think it might not be necessary to be able to `abort` the setter, and actually, maybe we should update our sections' implementation instead. It could be confusing.

Reply:
Yeah that seems very confusing. (I actually still don't understand it.)

Reply:
done